bittide/src/Bittide/ClockControl/ParseRegisters.hs

```diff
@@ -0,0 +1,83 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+module Bittide.ClockControl.ParseRegisters where
+
+import Prelude
+
+import Clash.Sized.Vector (listToVecTH)
+import Control.Monad.IO.Class (liftIO)
+import Data.List (isPrefixOf)
+import Data.List.Extra (trim)
+import Numeric (readHex)
+import System.FilePath ((</>))
+
+import Language.Haskell.TH
+
+import Paths_bittide (getDataFileName)
+
+type Triple = (Integer, Integer, Integer)
+type Lines = [String]
+
+-- | Parse a hex string into an integer.
+--
+-- >>> parseHex "1234"
+-- Right 4660
+-- >>> parseHex "foobar"
+-- Left "parseHex: could not parse: foobar. Got: [(15,\"oobar\")]"
+parseHex :: String -> Either String Integer
```

Comment:
Well.. the extension is CSV but it's not really :woozy_face:. Basically it looks like:

```
# bla bla
# Preamble
<some actual CSV>
# blabla
# Config
<some actual CSV>
# blabla
# Postamble
<some actual CSV>
```

Reply:
And it seems that you need that "Comment lines" to identify the different sections, otherwise I'd just remove them from the file.

Industry just using standards as they should ... :disappointed:

Reply:
yep yep yep

Reply:
I've reused the parsing logic now. I did simplify its implementation by a lot, I'll ping you on Slack.

Reply:
I haven't switched to cassava (or any other CSV parsing library), because:

* This isn't really CSV.. :unamused:
* I feel the code is still pretty simple/readable
* Over time we're going to move this to Rust, so I doubt getting this in a perfect state is really worth it